### ISSUE TITLE
NMS: the 3x3/5x5/7x7 fast paths and the general path disagree at image borders

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4239]
### Describe the bug

`NonMaximaSuppression2d` has two different border rules depending on the kernel size, and they
disagree.

* The hand-written `(3, 3)`, `(5, 5)` and `(7, 7)` paths allocate `mask = torch.zeros(...)` and fill
  only `mask[..., c:-c, c:-c]` with `c = (k - 1) // 2`. A pixel closer than `c` to the edge can never
  be a maximum.
* The general path (any other kernel size) replicate-pads and lets those pixels compete.

So `nms2d(x, (5, 5))` and `nms2d(x, (9, 9))` apply different rules to the same pixel, and neither is
documented. `(3, 3)` agrees with the general path only by accident: replicate padding puts a border
pixel's own value into its own window, so `x > x` suppresses it anyway. For `k >= 5` the rings at
distance `1 .. c-1` from the edge are where they diverge.

The general path's behaviour is also not "border pixels may compete" in any clean sense. Replicate
padding invents neighbours by duplicating edge values, which means:

* the outermost ring is always suppressed, because a pixel's own value is replicated next to it;
* rings `1 .. c-1` are judged against duplicated copies of *other* edge pixels, which both fabricates
  plateaus (suppressing genuine maxima) and hides genuine neighbours (admitting spurious ones).

### Reproduction steps

```python
import torch
from kornia.geometry.subpix import nms2d

x = torch.zeros(1, 1, 9, 9)
x[0, 0, 1, 1] = 1.0                       # one pixel, at distance 1 from the edge

print(nms2d(x, (5, 5), mask_only=True)[0, 0, 1, 1].item())   # False  -- explicit path
print(nms2d(x, (9, 9), mask_only=True)[0, 0, 1, 1].item())   # True   -- general path
```

An exhaustive single-pixel probe over 9x9, 12x17 and 20x20 images finds mismatches between the two
rules at every position in the `1 .. c-1` rings, for `k = 5` and `k = 7`, and none for `k = 3`.

### Expected behavior

One rule for every kernel size, and it should be documented.

Every detector in kornia already rejects the `(k - 1) // 2` border strip, by one route or another:

* `ALIKED` re-implements the rejection by hand right after calling `nms2d`
  (`kornia/feature/aliked/aliked.py`: `nms_scores[:, :, : self.radius, :] = 0`, and the three
  matching slices), with `radius = (kernel_size - 1) // 2` -- exactly the strip;
* `DISK` (`window_size=5`) and `XFeat` (`kernel_size=5`) take the explicit path, which rejects it;
* `MultiResolutionDetector` zeroes a 15 px border via `remove_borders` *before* the NMS, so its
  `nms_size=15` general-path call never sees those pixels.

Nothing in kornia relies on the replicate-padded behaviour, and a keypoint in that strip is not
usable downstream anyway -- its descriptor patch does not fit and `laf_is_inside_image` drops it.

### Environment

```
kornia 0.9.0rc1 | torch 2.14.0+cu130 | python 3.11.14 | OS Linux | device cpu
```

Written by Claude (Opus 5) on behalf of @ducha-aiki
